### ISSUE TITLE
Labeling of gt-tables is broken

### ISSUE THREAD

**Issue by snhansen** [quarto-dev/quarto-cli#5657]
### Bug description

With the newest Quarto version (1.4.84) on Windows, I now get an error when trying to label a table created with `gt()`. A minimal example:

	---
	format: html
	---

	# Title

	```{r}
	#| echo: false
	#| tbl-cap: "Caption."
	#| label: tbl-test
	library(gt)

	gt(mtcars)
	```

which results in this error:

	Error running filter C:/Users/au234616/AppData/Local/Programs/Quarto/share/filters/main.lua:
	...616/AppData/Local/Programs/Quarto/share/filters/main.lua:9301: attempt to call a string value (field 'html_table_caption')
	stack traceback:
			...616/AppData/Local/Programs/Quarto/share/filters/main.lua:9139: in local 'fn'
			...616/AppData/Local/Programs/Quarto/share/filters/main.lua:2120: in function <...616/AppData/Local/Programs/Quarto/share/filters/main.lua:2114>
			(...tail calls...)
			[C]: in ?
			[C]: in method 'walk'
			...616/AppData/Local/Programs/Quarto/share/filters/main.lua:178: in function 'run_emulated_filter'
			...616/AppData/Local/Programs/Quarto/share/filters/main.lua:566: in local 'callback'
			...616/AppData/Local/Programs/Quarto/share/filters/main.lua:577: in upvalue 'run_emulated_filter_chain'
			...616/AppData/Local/Programs/Quarto/share/filters/main.lua:610: in function <...616/AppData/Local/Programs/Quarto/share/filters/main.lua:607>
	stack traceback:
			...616/AppData/Local/Programs/Quarto/share/filters/main.lua:178: in function 'run_emulated_filter'
			...616/AppData/Local/Programs/Quarto/share/filters/main.lua:566: in local 'callback'
			...616/AppData/Local/Programs/Quarto/share/filters/main.lua:577: in upvalue 'run_emulated_filter_chain'
			...616/AppData/Local/Programs/Quarto/share/filters/main.lua:610: in function <...616/AppData/Local/Programs/Quarto/share/filters/main.lua:607>

### Checklist

- [X] Please include a minimal, fully reproducible example in a single .qmd file? Please provide the whole file rather than the snippet you believe is causing the issue.
- [X] Please [format your issue](https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us) so it is easier for us to read the bug report.
- [ ] Please document the RStudio IDE version you're running (if applicable), by providing the value displayed in the "About RStudio" main menu dialog?
- [X] Please document the operating system you're running. If on Linux, please provide the specific distribution.
- [ ] Please provide the output of `quarto check` so we know which version of quarto and its dependencies you're running.

**Comment by mcanouil** [comment]
Side note, should some tests be added for `gt`? As for [`knitr-tables-latex.qmd`](https://github.com/quarto-dev/quarto-cli/blob/main/tests/docs/crossrefs/knitr-tables-latex.qmd) and [`knitr-tables.qmd`](https://github.com/quarto-dev/quarto-cli/blob/main/tests/docs/crossrefs/knitr-tables.qmd)

**Comment by cderv** [comment]
> Side note, should some tests be added for gt?

I believe it is tracked by @rich-iannone and he plans to add some (if not already) on **gt** side to cover a vast variations of tables.

**Comment by cderv** [comment]
> It seems they are passing as we did not detect the issue before 🤔

This in fact gave us the hint that it is an issue with **gt** - we are not testing with **gt** 0.8  and now 0.9 is out. 

So issue is with **gt** 0.9 - @snhansen installing **gt** 0.8 solve the issue for me, so my guess is that you are using 0.9 too. 

Using Quarto 1.3 works with both version. 

@rich-iannone FYI  as related to **gt**

**Comment by snhansen** [comment]
@cderv: Ah, yes. Same behaviour here. Didn't occur to me that it could be related to gt.

**Comment by cscheid** [comment]
The issue is not with gt; it's that our tests were not covering `crossref/tables.lua:196`. It was a typo.

I've tested a local fix with gt 0.9.0, but I'm not sure how to make our test suite cover multiple R package versions.

**Comment by mcanouil** [comment]
What changed in gt that lead to Quarto to error out here?
The only way I know is to use `renv` profiles or similarly different lockfile which record different version of the packages.

**Comment by mcanouil** [comment]
The main "issue" is what packages and how many version to you want to test against?
Should Quarto really be compatible with possibly any versions of any packages? Or should any Quarto release worked with the latest release of the packages?

**Comment by cscheid** [comment]
I found a workaround; I used the 0.9.0 output directly on the qmd input. That triggered the bug independently of the library.

> Should Quarto really be compatible with possibly any versions of any packages? 

That's not the point here, because this isn't a GT bug. This was literally me using `foo()` instead of `foo` on a string value. It only happened that the bug triggered on 0.9.0 because the html formatting changed and it started exercising that specific code path.
